mantidproject/mantid
# New parameter in FindCenterOfMassPosition

For some measurements on biosans (cg3) there is a difficulty in finding the beam center. This PR is aimed at helping with that problem:
* Add parameter `IntegrationRadius` which allows for reducing the region of the detector to be used down to an area near the last iterations found center rather than the whole detector
* Additional unit test for data that required the new parameter
* Refactor of code to decouple private helper class and algorithm (lots of parameter/function renaming)
* Re-enable a unit test and fix typo in expected result
* Add check for results in existing doctest
* Refactor test to collect repetitive code into  a verify function

**To test:**

The data that showed the issue is a new test ``testCG3Data``. Verify that it makes sense.

*There is no associated issue,* but is the work for [EWM1746](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=1746)





---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
